Compilation error `error CS0579: Duplicate GeneratedCodeAttribute attribute` when used with webapi project

I have a webapi project created using `dotnet new webapi` which calls Orleans client and silo. The Orleans client and silo projects were working fine initially as a console application. But after inclusion of webapi project, I am getting a compilation error `error CS0579: Duplicate 'global::System.CodeDom.Compiler.GeneratedCodeAttribute' attribute`. The project and the associated bin-log can be found [here](https://drive.google.com/drive/folders/1k8pXqQ7l4VpaxWX7CW2_EtQC_YPtasSv?usp=sharing)

--- a/src/Orleans.CodeGeneration/CodeGeneratorCommon.cs
+++ b/src/Orleans.CodeGeneration/CodeGeneratorCommon.cs
@@ -45,15 +45,7 @@ namespace Orleans.CodeGenerator
                                 .AddArgumentListArguments(
                                     SF.AttributeArgument(asm.GetName().FullName.GetLiteralExpression()))).ToArray())
                     .WithTarget(SF.AttributeTargetSpecifier(SF.Token(SyntaxKind.AssemblyKeyword)));
-            var generatedCodeAttribute =
-                SF.AttributeList()
-                    .AddAttributes(
-                        SF.Attribute(typeof(GeneratedCodeAttribute).GetNameSyntax())
-                            .AddArgumentListArguments(
-                                SF.AttributeArgument(ToolName.GetLiteralExpression()),
-                                SF.AttributeArgument(RuntimeVersion.FileVersion.GetLiteralExpression())))
-                    .WithTarget(SF.AttributeTargetSpecifier(SF.Token(SyntaxKind.AssemblyKeyword)));
-            return generatedSyntax.Syntax.AddAttributeLists(generatedCodeAttribute, codeGenTargetAttributes);
+            return generatedSyntax.Syntax.AddAttributeLists(codeGenTargetAttributes);
         }
 
         internal static AttributeSyntax GetGeneratedCodeAttributeSyntax()

--- a/src/Orleans.CodeGeneration/RoslynCodeGenerator.cs
+++ b/src/Orleans.CodeGeneration/RoslynCodeGenerator.cs
@@ -154,6 +154,7 @@ namespace Orleans.CodeGenerator
         public string GenerateSourceForAssembly(Assembly input)
         {
             if (input.GetCustomAttribute<GeneratedCodeAttribute>() != null
+                || input.GetCustomAttribute<OrleansCodeGenerationTargetAttribute>() != null
                 || input.GetCustomAttribute<SkipCodeGenerationAttribute>() != null)
             {
                 return string.Empty;
@@ -652,6 +653,7 @@ namespace Orleans.CodeGenerator
             return !assembly.IsDynamic
                    && TypeUtils.IsOrleansOrReferencesOrleans(assembly)
                    && assembly.GetCustomAttribute<GeneratedCodeAttribute>() == null
+                   && assembly.GetCustomAttribute<OrleansCodeGenerationTargetAttribute>() == null
                    && assembly.GetCustomAttribute<SkipCodeGenerationAttribute>() == null;
         }
 

